<a href="https://colab.research.google.com/github/wahyudims/TokyoOlympic/blob/main/OlimpiadeTokyo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


# Import Modules

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go

# Import Datasets

In [3]:
teams=pd.read_excel('/content/gdrive/MyDrive/Colab Notebooks/StudiKasus/Teams.xlsx')
medals=pd.read_excel('/content/gdrive/MyDrive/Colab Notebooks/StudiKasus/Medals.xlsx')
genders=pd.read_excel('/content/gdrive/MyDrive/Colab Notebooks/StudiKasus/EntriesGender.xlsx')
coaches=pd.read_excel('/content/gdrive/MyDrive/Colab Notebooks/StudiKasus/Coaches.xlsx')
athletes=pd.read_excel('/content/gdrive/MyDrive/Colab Notebooks/StudiKasus/Athletes.xlsx')

In [4]:
print(teams.head())
print(medals.head())
print(genders.head())
print(coaches.head())
print(athletes.head())

      Name      Discipline                         NOC  Event
0  Belgium  3x3 Basketball                     Belgium    Men
1    China  3x3 Basketball  People's Republic of China    Men
2    China  3x3 Basketball  People's Republic of China  Women
3   France  3x3 Basketball                      France  Women
4    Italy  3x3 Basketball                       Italy  Women
   Rank                    Team/NOC  Gold  Silver  Bronze  Total  Rank by Total
0     1    United States of America    39      41      33    113              1
1     2  People's Republic of China    38      32      18     88              2
2     3                       Japan    27      14      17     58              5
3     4               Great Britain    22      21      22     65              4
4     5                         ROC    20      28      23     71              3
            Discipline  Female  Male  Total
0       3x3 Basketball      32    32     64
1              Archery      64    64    128
2  Artistic Gymn

# Genders Plot


*   Athletics discipline has the most participant (1072 Male and 969 Female)
*   Artistic Swimming and Rhythmic Gymnastic have no male participant
*   Cycling BMX Freestyle has the least participant



In [5]:
genders.sort_values(by='Total', ascending=True, inplace=True)
female = go.Bar(y=genders['Discipline'], 
                x=genders['Female'], 
                name='Female', 
                orientation='h', 
                text=genders['Female'], 
                textposition='outside', 
                hoverinfo='skip', 
                marker_color='pink')
male = go.Bar(y=genders['Discipline'], 
              x=genders['Male'], 
              name='Male', 
              orientation='h', 
              text=genders['Male'], 
              textposition='outside',
              hoverinfo='skip', 
              marker_color='blue')
data = [female, male]
layoutgender = go.Layout(title = {'text': '<b>Genders plot:</b> <br />Distribution accross all disciplines<br />', 'xanchor': 'center', 'x':0.5} ,
              xaxis= dict(title= 'Genders count',ticklen= 5,zeroline= False),
              hovermode="y", autosize=False, width=1000, height=2000,)
fig=go.Figure(data=data, layout=layoutgender)
fig.show()

# Athletes Plot
* United States of America sent the most athletes with 615 Athletes.
* There are at least 15 countries that only sent 2 athletes, mostly located in Oceania, Carribean, and Africa

In [6]:
adf=pd.DataFrame(athletes.groupby('NOC')['Discipline'].count())
adf.sort_values(by='Discipline', ascending=True, inplace=True)
layout = go.Layout(title = {'text': '<b>Athletes plot:</b> <br />Distribution accross all nations<br />', 'xanchor': 'center', 'x':0.5} ,
              xaxis= dict(title= 'Number of athletes',ticklen= 5,zeroline= False),
              hovermode="y", autosize=False, width=800, height=4000,)
fig=go.Figure(data=[go.Bar(y=adf.index, 
                           x=adf['Discipline'],
                           orientation='h', 
                           marker_color='rgb(100, 50, 150)')], layout=layout)
fig.show()

# Teams Plot
* Swimming has the most country participated with 113 countries participated
* Baseball/Softball has the least country participated with only 12 countries participated.

In [8]:
tdf=teams.groupby('Discipline')['Name'].count()
tdf=pd.DataFrame(tdf)
tdf.sort_values(by='Name', ascending=True, inplace=True)

                     Name
Discipline               
Baseball/Softball      12
Rhythmic Gymnastics    14
3x3 Basketball         16
Triathlon              18
Water Polo             22


In [9]:
layoutteams=go.Layout(title = {'text': '<b>Teams plot:</b> <br />Distribution accross all nations<br />', 
                                'xanchor': 'center', 'x':0.5} ,
                       yaxis= dict(title= 'Total',ticklen= 5,zeroline= False),
                      xaxis= dict(title= 'Nations participated',ticklen= 5,zeroline= False),
                       hovermode="y", autosize=False, width=800, height=500,)
fig=go.Figure(data=[go.Bar(y=tdf.index, 
                           x=tdf['Name'],
                           orientation='h', 
                           marker_color='rgb(200, 75, 150)')], 
              layout=layoutteams)
fig.show()

# Medals Plot
* United States of America collected the most medal with 39 Gold, 41 Silver, and 33 Bronze
* Oceanian is the only region who didn't collect any medals.

In [10]:
print(medals.info())
print(medals.head())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93 entries, 0 to 92
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Rank           93 non-null     int64 
 1   Team/NOC       93 non-null     object
 2   Gold           93 non-null     int64 
 3   Silver         93 non-null     int64 
 4   Bronze         93 non-null     int64 
 5   Total          93 non-null     int64 
 6   Rank by Total  93 non-null     int64 
dtypes: int64(6), object(1)
memory usage: 5.2+ KB
None
   Rank                    Team/NOC  Gold  Silver  Bronze  Total  Rank by Total
0     1    United States of America    39      41      33    113              1
1     2  People's Republic of China    38      32      18     88              2
2     3                       Japan    27      14      17     58              5
3     4               Great Britain    22      21      22     65              4
4     5                         ROC    20      28      2

In [11]:
medals.sort_values(by='Total', ascending=True, inplace=True)
print(medals.head())

    Rank              Team/NOC  Gold  Silver  Bronze  Total  Rank by Total
92    86  Syrian Arab Republic     0       0       1      1             77
62    63               Bermuda     1       0       0      1             77
76    77               Bahrain     0       1       0      1             77
77    77          Saudi Arabia     0       1       0      1             77
78    77             Lithuania     0       1       0      1             77


In [12]:
bronze = go.Bar(y=medals['Team/NOC'], 
                x=medals['Bronze'], 
                name='Bronze', 
                orientation='h', 
                # text=medals['Bronze'], 
                # textposition='outside', 
                # hoverinfo='skip', 
                marker_color='brown')
silver = go.Bar(y=medals['Team/NOC'], 
                x=medals['Silver'], 
                name='Silver', 
                orientation='h', 
                # text=medals['Silver'], 
                # textposition='outside', 
                # hoverinfo='skip', 
                marker_color='silver')
gold = go.Bar(y=medals['Team/NOC'], 
                x=medals['Gold'], 
                name='Gold', 
                orientation='h', 
                # text=medals['Gold'], 
                # textposition='outside', 
                # hoverinfo='skip', 
                marker_color='gold')
data = [bronze, silver, gold]
layoutmedals = go.Layout(title = {'text': '<b>Medals plot:</b> <br />Distribution accross all nations<br />', 'xanchor': 'center', 'x':0.5} ,
              yaxis= dict(title= 'Total',ticklen= 5,zeroline= False),
              hovermode="y", autosize=False, width=1000, height=10000,)
fig=go.Figure(data=data, layout=layoutmedals)
fig.show()